tests/dflowfm/test_mdu.py

```diff
@@ -388,3 +392,115 @@ def test_loading_fmmodel_model_with_both_ini_and_xyn_obsfiles(self):
+        program               = D-Flow FM     
+        version               = 1.2.100.66357 
+        autoStart             = 0             
+        pathsRelativeToParent = 0             
+        unknownkey            = something
+        """
+
+        tmp_mdu_path = tmp_path / "tmp.mdu"
+        tmp_mdu_path.write_text(tmp_mdu)
+
+        with patch(
+            "hydrolib.core.dflowfm.ini.models.INIBasedModel.Config"
+        ) as mock_config:
+            mock_config.extra = Extra.ignore
+
+            FMModel(filepath=tmp_mdu_path)
+            captured = capsys.readouterr()
+
+            section = "General"
+            name = "unknownkey"
+
+            assert (
+                f"Unknown keywords are detected in '{section}', these keywords will be dropped:"
+                in captured.out
+            )
+            assert name in captured.out
+
+    def test_mdu_unknown_keywords_loading_gives_message_for_missing_keyword2(
+        self, tmp_path, capsys
+    ):
+        tmp_mdu = """
+        [General]
+        fileVersion           = 1.09          
+        fileType              = modelDef      
+        program               = D-Flow FM     
+        version               = 1.2.100.66357 
+        autoStart             = 0             
+        pathsRelativeToParent = 0             
+        unknownkey            = something
+        """
+
+        tmp_mdu_path = tmp_path / "tmp.mdu"
+        tmp_mdu_path.write_text(tmp_mdu)
+
+        with patch(
+            "hydrolib.core.dflowfm.ini.models.INIBasedModel.Config"
+        ) as mock_config:
+            mock_config.extra = Extra.allow
+
+            FMModel(filepath=tmp_mdu_path)
+            captured = capsys.readouterr()
+
+            section = "General"
+            name = "unknownkey"
+
+            assert (
+                f"Unknown keywords are detected in '{section}', these keywords will be kept in memory but will have no validation:"
+                in captured.out
+            )
+            assert name in captured.out
+
+    def test_mdu_unknown_keywords_loading_does_not_give_message_on_exclude_fields(
+        self, tmp_path, capsys
+    ):
+        tmp_mdu = """
+        [General]
+        fileVersion           = 1.09          
+        fileType              = modelDef      
+        program               = D-Flow FM     
+        version               = 1.2.100.66357 
+        autoStart             = 0             
+        pathsRelativeToParent = 0             
+        unknownkey            = something
+        """
+
+        tmp_mdu_path = tmp_path / "tmp.mdu"
+        tmp_mdu_path.write_text(tmp_mdu)
+
+        FMModel(filepath=tmp_mdu_path)
+        captured = capsys.readouterr()
+
+        excluded_fields = ["comments", "datablock", "_header"]
+        for excluded_field in excluded_fields:
+            assert excluded_field not in captured.out
+
+    def test_mdu_unknown_keywords_allow_extra_setting_field_gives_message(self, capsys):
```

Comment:
In my opinion there is no need to check all the different options for `Extra.<allow/forbid/ignore>`. The user should not edit the values we have set, since it's not a public setting. If they decide to change it anyway, we can't offer support/validation.